### ISSUE TITLE
`xor` is not xor

### ISSUE THREAD

**Issue by whonore** [tokahuke/yaque#17]
`xor` in [header.rs](https://github.com/tokahuke/yaque/blob/master/src/header.rs#L76) does not actually compute the exclusive or of `a` and `b`. Instead it checks that `a` and `b` are either both true or both false, which is equivalent to `a == b`. I don't understand what `decode` does well enough to know if this is the intended behavior or not, but either the name or the implementation is currently wrong. If you do actually need `xor` there's the `^` operator and the [`BitXor` trait](https://doc.rust-lang.org/std/ops/trait.BitXor.html).

**Comment by NinoScript** [comment]
You're right, that code is the opposite of a `xor`, it's an `xnor`: `!(a ^ b)`
